File: PAT-B/python/1002.py
class solution(object):
	num=['ling','yi','er','san','si','wu','liu','qi','ba','jiu']
	def print_t(self,n):
		n=int(n)
		sum = 0
		while(n!=0):
			sum+=n%10
			n=n//10
		temp=sum%10
		str=' '+self.num[temp]
		sum=sum//10	
		while(sum):
			temp=sum%10
			str=' '+self.num[temp]+str
			sum=sum//10
		str=str[1:]
		print(str,end='')
if __name__=='__main__'	:	
	#print(__name__)
	m=solution()
	a=input()
	m.print_t(a)

File: PAT-B/1002.c
#include<stdio.h>
void print(int i);
char c[101];
char n[10][20]={"ling","yi","er","san","si","wu","liu","qi","ba","jiu"};
int main()
{
    scanf("%s",c);
    int i=0;
    int sum = 0;
    while(c[i]!='\0'){
        sum = sum +c[i]-'0';
        i++;
    }
    print(sum);
    return 0;
}
void print(int i)
{
    if(i/10==0){
        printf("%s",n[i%10]);
    }
    else{
        print(i/10);
        printf(" %s",n[i%10]);
    }
}
